lib/isaac/isaac_fusion_extension.hpp

```diff
@@ -22,6 +22,22 @@
 #include <boost/mpl/int.hpp>
 #include <boost/mpl/size.hpp>
 
+namespace std
+{
+    /* @bug ISAAC is working with boost fusion lists, e.g. boost::fusion::cons<>
+     * these types are not trivially copyable but are passed as parameters to the kernel.
+     * Since alpaka 0.9.0 the requirement to be trivially copyable is correct enforced,
+     * with the result that these types can not be used as kernel parameter.
+     * This trait specialization is a temporary and risky workaround until boost fusion is
+     * removed from ISAAC.
+     * @attention This workaround can have bad effects on other libraries which using boost fusion and ISAAC.
+     */
```

Comment:
I can't judge how forced are you into this decision, so practically it's maybe least bad. However, at least `<type_traits>` must be included (otherwise the trait is not even defined), and it's not just `can have bad effects`, specializing `is_trivially_copyable<>` is undefined behavior [according to cppreference](https://en.cppreference.com/w/cpp/types/is_trivially_copyable)

Reply:
Perhaps for cases like that we can add a "backdoor" trait for alpaka so that we can whitelist certain types and make alpaka trust they are fine to be used in kernels without checking for trivially copyable. The difference is, we would here specialize an alpaka trait instead of `std::` one, and so not rist undefined behavior.

Reply:
I do not know if a alpaka whitelist trait is required. In general, I do not like what I did but I saw no other way to get this fast in.
Adding the feature first to alpaka and waiting months until it is propagated to ISAAC was no solution for me.

Reply:
`<type_traits>` You are right I missed the include.

Reply:
> Adding the feature first to alpaka and waiting months until it is propagated to ISAAC was no solution for me.

I meant to consider adding it for alpaka, not waiting right now until it's implemented and released in alpaka